minor warnings under firefox / jslint?

when using chai, firefox has several minor complaints

```
console.log: shield-studies-addon-utils: [JavaScript Warning: "TypeError: variable obj redeclares argument" {file: "resource://gre/modules/commonjs/toolkit/loader.js -> resource://shield-studies-addon-utils/node_modules/chai/lib/chai/utils/expectTypes.js" line: 26 column: 6 source: "  var obj = flag(obj, 'object');
"}]
console.log: shield-studies-addon-utils: [JavaScript Warning: "TypeError: variable msg redeclares argument" {file: "resource://gre/modules/commonjs/toolkit/loader.js -> resource://shield-studies-addon-utils/node_modules/chai/lib/chai/assertion.js" line: 105 column: 10 source: "      var msg = util.getMessage(this, arguments)
"}]

```

These would have been cause by using `eslint` with `no-redeclare`:  <http://eslint.org/docs/rules/no-redeclare>

## Solution

- I will happily submit a patch in about 2 minutes to fix it :)
- perhaps consider incorporating an `eslint` step in the build / deploy process.

Thanks!

GL

```
50278 glind ~/gits/chai [git:775-no-redeclare?]$ eslint "lib/**"

/Users/glind/gits/chai/lib/chai/assertion.js
  107:11  error  'msg' is already defined  no-redeclare

/Users/glind/gits/chai/lib/chai/core/assertions.js
  1763:11  error  'subject' is already defined  no-redeclare

/Users/glind/gits/chai/lib/chai/utils/expectTypes.js
  26:7  error  'obj' is already defined  no-redeclare

/Users/glind/gits/chai/lib/chai/utils/inspect.js
  135:9  error  'name' is already defined        no-redeclare
  136:9  error  'nameSuffix' is already defined  no-redeclare

✖ 5 problems (5 errors, 0 warnings)
```

--- a/lib/chai/assertion.js
+++ b/lib/chai/assertion.js
@@ -104,7 +104,7 @@ module.exports = function (_chai, util) {
     if (true !== config.showDiff) showDiff = false;
 
     if (!ok) {
-      var msg = util.getMessage(this, arguments)
+      msg = util.getMessage(this, arguments)
         , actual = util.getActual(this, arguments);
       throw new AssertionError(msg, {
           actual: actual

--- a/lib/chai/core/assertions.js
+++ b/lib/chai/core/assertions.js
@@ -1753,14 +1753,14 @@ module.exports = function (chai, _) {
     var contains = flag(this, 'contains');
     var ordered = flag(this, 'ordered');
 
-    var failMsg, failNegateMsg, lengthCheck;
+    var subject, failMsg, failNegateMsg, lengthCheck;
 
     if (contains) {
-      var subject = ordered ? 'an ordered superset' : 'a superset';
+      subject = ordered ? 'an ordered superset' : 'a superset';
       failMsg = 'expected #{this} to be ' + subject + ' of #{exp}';
       failNegateMsg = 'expected #{this} to not be ' + subject + ' of #{exp}';
     } else {
-      var subject = ordered ? 'ordered members' : 'members';
+      subject = ordered ? 'ordered members' : 'members';
       failMsg = 'expected #{this} to have the same ' + subject + ' as #{exp}';
       failNegateMsg = 'expected #{this} to not have the same ' + subject + ' as #{exp}';
     }

--- a/lib/chai/utils/expectTypes.js
+++ b/lib/chai/utils/expectTypes.js
@@ -23,7 +23,7 @@ var flag = require('./flag');
 var type = require('type-detect');
 
 module.exports = function (obj, types) {
-  var obj = flag(obj, 'object');
+  obj = flag(obj, 'object');
   types = types.map(function (t) { return t.toLowerCase(); });
   types.sort();
 

--- a/lib/chai/utils/inspect.js
+++ b/lib/chai/utils/inspect.js
@@ -99,6 +99,8 @@ function formatValue(ctx, value, recurseTimes) {
   var visibleKeys = getEnumerableProperties(value);
   var keys = ctx.showHidden ? getProperties(value) : visibleKeys;
 
+  var name, nameSuffix;
+
   // Some type of object without properties can be shortcutted.
   // In IE, errors have a single `stack` property, or if they are vanilla `Error`,
   // a `stack` plus `description` property; ignore those for consistency.
@@ -107,8 +109,8 @@ function formatValue(ctx, value, recurseTimes) {
       (keys.length === 2 && keys[0] === 'description' && keys[1] === 'stack')
      ))) {
     if (typeof value === 'function') {
-      var name = getName(value);
-      var nameSuffix = name ? ': ' + name : '';
+      name = getName(value);
+      nameSuffix = name ? ': ' + name : '';
       return ctx.stylize('[Function' + nameSuffix + ']', 'special');
     }
     if (isRegExp(value)) {
@@ -132,8 +134,8 @@ function formatValue(ctx, value, recurseTimes) {
 
   // Make functions say that they are functions
   if (typeof value === 'function') {
-    var name = getName(value);
-    var nameSuffix = name ? ': ' + name : '';
+    name = getName(value);
+    nameSuffix = name ? ': ' + name : '';
     base = ' [Function' + nameSuffix + ']';
   }
 

